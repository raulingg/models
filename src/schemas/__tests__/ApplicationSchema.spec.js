const mongoose = require('mongoose/browser');
const ApplicationSchema = require('../ApplicationSchema')(mongoose);
const CohortSchema = require('../CohortSchema')(mongoose);

describe('ApplicationSchema', () => {
  it('should fail validation when fields are missing', () => {
    const doc = new mongoose.Document({}, ApplicationSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when some of fields are invalid', () => {
    const cohort = new mongoose.Document({}, CohortSchema);
    const doc = new mongoose.Document(
      {
        cohort: cohort._id,
        email: 'test@example.com',
        identificationNumber: '45487844',
        name: 'Test',
        familyName: 'Laboratoria',
        appliedAt: 'notDate',
        referralSource: 'Facebook',
        passed: 'notBoolean',
        mobileNumber: '45254778',
      },
      ApplicationSchema,
    );

    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should pass validation when fields are valid', () => {
    const cohort = new mongoose.Document({}, CohortSchema);
    const doc = new mongoose.Document(
      {
        cohort: cohort._id,
        email: 'test@example.com',
        identificationNumber: '45487844',
        name: 'Test',
        familyName: 'Laboratoria',
        appliedAt: new Date('2019-06-06'),
        referralSource: 'Facebook',
        passed: true,
        mobileNumber: '45254778',
        isReturningApplicant: false,
      },
      ApplicationSchema,
    );

    return doc.validate();
  });
});
