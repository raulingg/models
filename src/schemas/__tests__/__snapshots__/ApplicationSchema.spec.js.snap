// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationSchema should fail validation when fields are missing 1`] = `
Object {
  "cohort": [ValidatorError: Path \`cohort\` is required.],
  "email": [ValidatorError: Path \`email\` is required.],
  "familyName": [ValidatorError: Path \`familyName\` is required.],
  "identificationNumber": [ValidatorError: Path \`identificationNumber\` is required.],
  "mobileNumber": [ValidatorError: Path \`mobileNumber\` is required.],
  "name": [ValidatorError: Path \`name\` is required.],
  "passed": [ValidatorError: Path \`passed\` is required.],
}
`;

exports[`ApplicationSchema should fail validation when some of fields are invalid 1`] = `
Object {
  "appliedAt": [CastError: Cast to Date failed for value "notDate" at path "appliedAt"],
  "passed": [CastError: Cast to Boolean failed for value "notBoolean" at path "passed"],
}
`;
